ansible/roles/fanout/tasks/mlnx/upgrade_onyx.yml

```diff
@@ -0,0 +1,86 @@
+##############################################################################################
+### sub-playbook to perform Onyx to Onyx OS upgrade on fanout switch
+### To run this play seperately it requires image_url parameter.
+### E.g.
+### ansible-playbook -i lab fanout.yml -l ${FANOUT} -e image_url=http://some-http-server/onyx_os_image --become --tags upgrade -vvvv
+################################################################################################
+- name: Check if image_url is defined
+  fail:
+      msg: "please pass 'image_url' variable"
+  when: image_url is not defined
+
+- name: Define fanout image location
+  set_fact:
+    fanout_img_path: "/var/opt/tms/images"
+
+- name: Get current running Onyx OS version
+  include_tasks: get_onyx_os_version.yml
+
+- name: Print currnetly running Onyx OS version
+  debug:
+    msg: "Currently running Onyx OS version is \"{{ onyx_version }}\""
+
+- name: Download Onyx OS image on fanout switch
+  include_tasks: download_copy_image_fanout.yml
+  vars:
+    image_url: "{{ image_url }}"
+    fanout_dest: "{{ fanout_img_path }}/{{ image_url | basename }}"
+
+- name: Upgrade Onyx OS on fanout switch
+  action: apswitch template=mlnx_upgrade_onyx.j2
+  connection: switch
+  register: installation_output
+  args:
+    login: "{{ switch_login['MLNX-OS'] }}"
+    timeout: 10000
+
+- name: Print installation output for later debugging installation failures
+  debug:
+    var: installation_output
+
+- name: Wait for fanout to shutdown
+  become: false
+  local_action: wait_for
+  args:
+    host: "{{ ansible_host }}"
+    port: 22
+    state: absent
+    search_regex: "OpenSSH_[\\w\\.]+"
+    delay: 10
+    timeout: 180
+
+- name: Wait for fanout to come back
+  become: false
+  local_action: wait_for
+  args:
+    host: "{{ ansible_host }}"
+    port: 22
+    state: started
+    search_regex: "OpenSSH_[\\w\\.]+"
+    delay: 10
+    timeout: 180
+
+# After SSH is started, system is not ready to function
+- name: Pause for two more minutes till Onyx OS modules get configured
+  pause:
+    minutes: 2
+
+- name: Save old Onyx OS version
+  set_fact:
+    old_onyx_verison: "{{ onyx_version }}"
+
+- name: Get current running Onyx OS version
+  include_tasks: get_onyx_os_version.yml
+
+- name: Report currently running Onyx OS version
+  debug:
+    msg: "Currently running Onyx OS version is \"{{ onyx_version }}\""
+
+# Basic sanity check to verify that Onyx hasn't rolled back to previous image.
```

Comment:
IMO Onyx will not go to the burn flow in such case. do you get any prompt?

Reply:
In case of configuration migration failure Onyx by default will fallback to previous image.
I think this is a good behavior since in case of failure the fanout will recover itself so user will not break regression tests and we discussed this with Nazarii.
In such scenario we want to let the user know that there was a problem and it is done by checking whether currently running onyx version is a desired onyx version.